src/x/net/http/client.go

```diff
@@ -34,14 +39,15 @@ type HTTPClientOptions struct {
 	IdleConnTimeout    time.Duration `yaml:"idleConnTimeout"`
 	MaxIdleConns       int           `yaml:"maxIdleConns"`
 	DisableCompression bool          `yaml:"disableCompression"`
+	Proxy              ProxyFunc     `yaml:"proxy"`
 }
 
 // NewHTTPClient constructs a new HTTP Client.
 func NewHTTPClient(o HTTPClientOptions) *http.Client {
 	return &http.Client{
 		Timeout: o.RequestTimeout,
 		Transport: &http.Transport{
-			Proxy: http.ProxyFromEnvironment,
+			Proxy: o.Proxy,
```

Comment:
Can we keep this as `http.ProxyFromEnvironment` unless the user specifies a non-nil proxy function?

Reply:
So I set the default in `DefaultHTTPClientOptions` to `http.ProxyFromEnvironment`, do we want a check in addition to that? I'm happy to add one, it just seems that odd that we would check this field only.

Reply:
Yeah I'd feel more comfortable in case someone constructs `HTTPClientOptions` from scratch (not with `Default...`), since we've now changed what used to be an explicit default to a potentially nil value.

Reply:
👍 fair point, will update.